### PR TITLE
sui 1.17.3 (new formula)

### PR DESCRIPTION
**Pull request by johnjmartin**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

This is the cli for the Sui blockchain. It's used for publishing and testing Move packages, key management, staking, and a variety of other use cases.

tested on macOS 14.3 (ARM) and Ubuntu 22.04 LTS x64

### CODE REVIEW

**Comment by johnjmartin** [comment]
Looks like ci is failing due to the `CI / macOS 12-x86_64` build timing out, ours tends to take a ~50 minutes on github's `macos-latest-xl` runners: https://github.com/MystenLabs/sui/actions/runs/7548915905/job/20551926879

Can we try bumping the timeout on the CI job for this case, or should I look into speeding up our macOS builds?

File: Formula/s/sui.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  on_linux do
+    depends_on "llvm" => :build
+  end
+
+  def install
+    system "cargo", "install", *std_cargo_args(path: "crates/sui")
+  end
+
+  test do
+    system "#{bin}/sui", "keytool", "list"
```
Comment (chenrui333) [inline Formula/s/sui.rb:26]: can we also assert the response?

Comment (johnjmartin) [reply Formula/s/sui.rb]: There's no response if there's no available keystore for the tool, so I've adjusted the test to use a placeholder keystore

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
@johnjmartin, thanks for your first contribution to Homebrew! :tada: 🥇

Without awesome contributors like you, it would be impossible to maintain Homebrew to the high level of quality users have come to expect. Thank you!!!!

**Pull request merged**